lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1339,357 +1339,359 @@ def cmd_sessions_help
+      else
+        # Parse the command options
+        @@sessions_opts.parse(args) do |opt, idx, val|
+          case opt
+          when "-q", "--quiet"
+            quiet = true
+          # Run a command on all sessions, or the session given with -i
+          when "-c", "--command"
+            method = 'cmd'
             cmds << val if val
-        # Display the list of inactive sessions
-        when "-d", "--list-inactive"
-          show_inactive = true
-          method = 'list_inactive'
-        when "-x", "--list-extended"
-          show_extended = true
-        when "-v", "--list-verbose"
-          verbose = true
-        # Do something with the supplied session identifier instead of
-        # all sessions.
-        when "-i", "--interact"
-          sid = val
-        # Display the list of active sessions
-        when "-l", "--list-active"
-          show_active = true
-          method = 'list'
-        when "-k", "--kill"
-          method = 'kill'
-          sid = val || false
-        when "-K", "--kill-all"
-          method = 'killall'
-        # Run a script or module on specified sessions
-        when "-s", "--script"
-          unless script
-            method = 'script'
-            script = val
-          end
-        # Upload and exec to the specific command session
-        when "-u", "--upgrade"
-          method = 'upexec'
-          sid = val || false
-        # Search for specific session
-        when "-S", "--search"
-          search_term = val
-        # Display help banner
-        when "-h", "--help"
-          cmd_sessions_help
-          return false
-        when "-t", "--timeout"
-          if val.to_s =~ /^\d+$/
-            response_timeout = val.to_i
+          when "-C", "--meterpreter-command"
+              method = 'meterp-cmd'
+              cmds << val if val
+          # Display the list of inactive sessions
+          when "-d", "--list-inactive"
+            show_inactive = true
+            method = 'list_inactive'
+          when "-x", "--list-extended"
+            show_extended = true
+          when "-v", "--list-verbose"
+            verbose = true
+          # Do something with the supplied session identifier instead of
+          # all sessions.
+          when "-i", "--interact"
+            sid = val
+          # Display the list of active sessions
+          when "-l", "--list-active"
+            show_active = true
+            method = 'list'
+          when "-k", "--kill"
+            method = 'kill'
+            sid = val || false
+          when "-K", "--kill-all"
+            method = 'killall'
+          # Run a script or module on specified sessions
+          when "-s", "--script"
+            unless script
+              method = 'script'
+              script = val
+              # Assume any parameter after the script name is a flag/argument we want to pass to the script itself.
+              args[args.index(val) + 1..args.length - 1].each { |a| extra << a }
```

Comment:
Can we use `idx` here?

```suggestion
              args[idx + 1..args.length - 1].each { |a| extra << a }
```